pkg/user/tx_client_test.go

```diff
@@ -447,3 +452,241 @@ func setupEstimatorService(t *testing.T) *mockEstimatorServer {
+			},
+			expectedErr: nil,
+		},
+		{ // Secondary Success
+			setupMocks: func(t *testing.T) ([]*grpctest.MockTxService, []*grpc.ClientConn, []func()) {
+				mockSvc1 := &grpctest.MockTxService{ // Primary fails after delay
+					BroadcastHandler: func(ctx context.Context, req *sdktx.BroadcastTxRequest) (*sdktx.BroadcastTxResponse, error) {
+						time.Sleep(50 * time.Millisecond)
+						return nil, errMock1
+					}}
+				mockSvc2 := &grpctest.MockTxService{ // Secondary succeeds quickly
+					BroadcastHandler: func(ctx context.Context, req *sdktx.BroadcastTxRequest) (*sdktx.BroadcastTxResponse, error) {
+						time.Sleep(10 * time.Millisecond)
+						return &sdktx.BroadcastTxResponse{TxResponse: &sdk.TxResponse{Code: abci.CodeTypeOK, TxHash: "HASH2"}}, nil
+					}}
+				mockSvc3 := &grpctest.MockTxService{ // Tertiary should be cancelled
+					BroadcastHandler: func(ctx context.Context, req *sdktx.BroadcastTxRequest) (*sdktx.BroadcastTxResponse, error) {
+						select {
+						case <-time.After(500 * time.Millisecond):
+							return nil, errors.New("mock3 should have been cancelled")
+						case <-ctx.Done():
+							return nil, ctx.Err()
+						}
+					}}
+				conn1, stop1 := grpctest.StartMockServer(t, mockSvc1)
+				conn2, stop2 := grpctest.StartMockServer(t, mockSvc2)
+				conn3, stop3 := grpctest.StartMockServer(t, mockSvc3)
+				return []*grpctest.MockTxService{mockSvc1, mockSvc2, mockSvc3}, []*grpc.ClientConn{conn1, conn2, conn3}, []func(){stop1, stop2, stop3}
+			},
+			expectedErr: nil,
+		},
+		{ // All Fail
+			setupMocks: func(t *testing.T) ([]*grpctest.MockTxService, []*grpc.ClientConn, []func()) {
+				mockSvc1 := &grpctest.MockTxService{BroadcastHandler: func(ctx context.Context, req *sdktx.BroadcastTxRequest) (*sdktx.BroadcastTxResponse, error) {
+					return nil, errMock1
+				}}
+				mockSvc2 := &grpctest.MockTxService{BroadcastHandler: func(ctx context.Context, req *sdktx.BroadcastTxRequest) (*sdktx.BroadcastTxResponse, error) {
+					return nil, errMock2
+				}}
+				mockSvc3 := &grpctest.MockTxService{BroadcastHandler: func(ctx context.Context, req *sdktx.BroadcastTxRequest) (*sdktx.BroadcastTxResponse, error) {
+					return nil, errMock3
+				}}
+				conn1, stop1 := grpctest.StartMockServer(t, mockSvc1)
+				conn2, stop2 := grpctest.StartMockServer(t, mockSvc2)
+				conn3, stop3 := grpctest.StartMockServer(t, mockSvc3)
+				return []*grpctest.MockTxService{mockSvc1, mockSvc2, mockSvc3}, []*grpc.ClientConn{conn1, conn2, conn3}, []func(){stop1, stop2, stop3}
+			},
+			expectedErr: errMock1, // Expect the first error returned (or any, depends on race)
+		},
+		{ // Context Deadline
+			setupMocks: func(t *testing.T) ([]*grpctest.MockTxService, []*grpc.ClientConn, []func()) {
+				mockHandler := func(ctx context.Context, req *sdktx.BroadcastTxRequest) (*sdktx.BroadcastTxResponse, error) {
+					select {
+					case <-time.After(200 * time.Millisecond):
```

Comment:
set a short deadline in the caller's ctx and wait til after the deadline, then return ctx err.